### DISCUSSION TITLE
Pre Diffusing variable

### DISCUSSION THREAD

**Discussion by gj19866** [idaholab/moose#25817] [Q&A General] [answered]
Hi there,

What would be the best method to pre diffuse a random variable before the main solve starts?

Would this best be done with a multiapp system, or by setting a kernel only to be active before/after a chose time?

Are there any examples of this?

Thanks
Gillian

**Comment by GiudGiud** [comment, marked as answer]
Hello

Both options are possible.
If the random variable is not a main system varaible for the main solve then you will have to use a multiapp. 

also if in the main solve there are more variables than the diffused one, the first part with just diffusion and nothing for the other variable, you will have to be kernel to use a NullKernel for the other variables. 

Guillaume

**Reply by gj19866** [reply]
Hi Guillaume,

The method I have landed on is to just allow the diffusion to happen alongside the normal solve, as I can change the diffusion coefficient so that it happens on a time scale much shorter than that of the rest of the system.

Thank you for all your help.

Best wishes,
Gillian
